modal/_container_entrypoint.py

```diff
@@ -380,27 +395,31 @@ async def complete_call(self, started_at):
         self.calls_completed += 1
         self._semaphore.release()
 
-    async def enqueue_output(self, input_id, started_at: float, output_index: int, data):
+    async def enqueue_output(self, input_id, started_at: float, output_index: int, data: Any, data_format: int) -> None:
         await self._enqueue_output(
             input_id,
             started_at=started_at,
             gen_index=output_index,
+            data_format=data_format,
             status=api_pb2.GenericResult.GENERIC_STATUS_SUCCESS,
-            data=self.serialize(data),
+            data=self.serialize_data_format(data, data_format),
```

Comment:
this would start writing ASGI serialized format before we added support for it on the other side? i'm not quite following

Reply:
(nvm, just saw the comment that this will be broken up)

Reply:
Yep! The rest of this file is just plumbing for `serialize_data_format`